### ISSUE TITLE
TypeScript watcher consumes large amount of CPU when idle.

### ISSUE THREAD

**Issue by MicahZoltu** [Microsoft/TypeScript#17506]
**TypeScript Version:**  2.4.2

**Expected behavior:**
When I am not touching any files in my project, TSC with `watch: true` configured will not consume much (if any) CPU.

**Actual behavior:**
When I am not touching any files in my project, TSC with `watch: true` configured consumes a notable amount of CPU on my laptop, enough to keep the fan on consistently.

[This blog post](https://medium.com/@julioromano/writing-typescript-on-a-laptop-this-might-improve-your-battery-life-f503dd16f019) suggests that the culprit is a [recent change to polling for changes](https://github.com/Microsoft/TypeScript/pull/8196/).

**Notes:**
I am using VSCode with the built-in TypeScript support, the default TypeScript build task and a tsconfig.json like:
```
{
	"compilerOptions": {
		"target": "es2017",
		"module": "commonjs",
		"sourceMap": true,
		"outDir": "output",
		"rootDir": "source",
		"paths": {
			"*": [
				"./typings/*"
			]
		},
		"emitDecoratorMetadata": true,
		"experimentalDecorators": true,
		"strictNullChecks": true,
		"alwaysStrict": true,
		"allowJs": true,
		"watch": true
	}
}
```

**Comment by Fistynuts** [comment]
I'm seeing something similar with tsc 2.3.4. 

With polling enabled, tsc -w -p <folder> takes up 10-15% CPU on our project files.
With polling disabled, the same command uses virtually no CPU if no files are changing.

Windows 10, Core i5.

**Comment by mhegazy** [comment]
Should be fixed in #17269. Please give typescript@next a try and let us know if you are still running into issues.
